Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -2,178 +2,251 @@ import Foundation
-    private var eventLogging: EventLogging?
+    private let dataProvider: CrashLoggingDataProvider
+    private let eventLogging: EventLogging?
 
-    /// Thread-safe single initialization
-    fileprivate static let threadSafeDispatchQueue = DispatchQueue(label: Bundle.main.bundleIdentifier ?? "tracks" + "-crash-logging-queue")
-    fileprivate static var _isStarted = false
-    internal static var isStarted: Bool {
-        get {
-            return threadSafeDispatchQueue.sync { _isStarted }
-        }
-        set {
-            threadSafeDispatchQueue.sync { _isStarted = newValue }
-        }
+    public init(dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+        self.dataProvider = dataProvider
+        self.eventLogging = eventLogging
     }
 
-    /**
-     Initializes the crash logging system.
-
-     - Parameters:
-     - dataProvider: An object that will provide any required data to the crash logging system.
-
-     - SeeAlso: CrashLoggingDataProvider
-     */
-    public static func start(withDataProvider dataProvider: CrashLoggingDataProvider, eventLogging: EventLogging? = nil) {
+    public func start() throws -> CrashLogging {
 
-        // Only allow initializing this system once
-        guard !isStarted else { return }
-        isStarted = true
+        /// Validate the DSN ourselves before initializing, because the SentrySDK silently prints the error to the log instead of telling us if the DSN is valid
+        _ = try SentryDsn(string: self.dataProvider.sentryDSN)
 
-        // Store the data provider and event logging subsystem for future use
-        sharedInstance.dataProvider = dataProvider
-        sharedInstance.eventLogging = eventLogging
+        SentrySDK.start { options in
+            options.dsn = self.dataProvider.sentryDSN
+            options.debug = true // Helpful to see what's going on
 
-        // Create a Sentry client and start crash handler
-        do {
-            Client.shared = try Client(dsn: dataProvider.sentryDSN)
-            try Client.shared?.startCrashHandler()
+            options.environment = self.dataProvider.buildType
+            options.releaseName = self.dataProvider.releaseName
+            options.enableAutoSessionTracking = self.dataProvider.shouldEnableAutomaticSessionTracking
 
-            // Automatically track screen transitions
-            if dataProvider.shouldEnableAutomaticBreadcrumbTracking {
-                Client.shared?.breadcrumbs.maxBreadcrumbs = 500 // Store lots of breadcrumbs to trace errors
-                Client.shared?.enableAutomaticBreadcrumbTracking()
-            }
+            options.beforeSend = self.beforeSend
 
-            // Override event serialization to append the logs, if needed
-            Client.shared?.beforeSerializeEvent = sharedInstance.beforeSerializeEvent
-            Client.shared?.shouldSendEvent = sharedInstance.shouldSendEvent
+            /// Attach stack traces to non-fatal errors
```

Comment:
```suggestion
            // Attach stack traces to non-fatal errors
```

Same rationale as above.